nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/mail/Mailer.java

```diff
@@ -147,9 +148,14 @@ public void setServer(String host, String port, boolean ssl) {
      *
      * @param user
      * @param pass
+     *
+     * @deprecated since 11.1, use {@code mail.${protocol}.user} and {@code mail.${protocol}.password} instead
```

Comment:
i am wondering about specifying `${protocol}` why not `smtp` directly as we send emails and [here](https://github.com/nuxeo/nuxeo/pull/3599/commits/96dace808c89f19ad1f51b7855ef8b532fba4388#diff-4d64f7f697b6034a5aba9f4eb98766d1R157)

Reply:
In fact the whole class leverages the _smtp_ protocol, you can have a look to the constructor. You can build a `javax.mail.Session` with `Mailer` leveraging another protocol, but you can't use its API to use another protocol. 

That's why I wanted to deprecate `Mailer` due to how its APIs behave, like the one you mention. But, I can't do it now because `Mailer` also handles sending a mail.

In then end, I'm in favor to use the new builder to retrieve a Session and, on next ticket about mail, see if we can rework more. Wdyt?